### ISSUE TITLE
Android port does not compile

### ISSUE THREAD

**Issue by LunarWatcher** [LWJGL/lwjgl3#354]
### Environment

* LWJGL version: `3` 
* LWJGL build #: `3.1.2`
* Java version: `1.8`
* Platform: Windows, compiling for Android
* Module: Android
* NDK: Revision 16 

### Description

I did everything it said on the [setup page](https://github.com/LWJGL/android-test), and after getting through the etc errors I finally got down to the native compile (c/c++ files). When compiling, I'm getting this error:

```
compile-native-platform:
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\util\xxhash\org_lwjgl_util_xxhash_LibXXHash.c:6:
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler]  #include <inttypes.h>
 [Compiler]                       ^
 [Compiler] compilation terminated.
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\nuklear\org_lwjgl_nuklear_Nuklear.c:10:
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\util\par\org_lwjgl_util_par_LibPar.c:6:
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\openvr\org_lwjgl_openvr_VRCompositor.c:6:
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\stb\org_lwjgl_stb_LibSTB.c:6:
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\util\xxhash\org_lwjgl_util_xxhash_XXHash.c:9:
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler]  #include <inttypes.h>
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler]  #include <inttypes.h>
 [Compiler]  #include <inttypes.h>
 [Compiler]  #include <inttypes.h>
 [Compiler]                       ^
 [Compiler] compilation terminated.
 [Compiler]                  from D:\programming\lwjgl3\lwjgl3\modules\core\src\generated\c\system\android\org_lwjgl_system_android_ANativeWindow.c:6:
 [Compiler]                       ^
 [Compiler] ../../../../modules\core\src\main\c/system/linux/LinuxConfig.h:7:22: fatal error: inttypes.h: No such file or directory
 [Compiler]  #include <inttypes.h>
 [Compiler]                       ^
 [Compiler] compilation terminated.
 [Compiler]  #include <inttypes.h>
 [Compiler] compilation terminated.
 [Compiler]                       ^
 [Compiler] compilation terminated.
 [Compiler]                       ^
 [Compiler]                       ^
 [Compiler] compilation terminated.
 [Compiler] compilation terminated.
 [Compiler] In file included from ../../../../modules\core\src\main\c/system/common_tools.h:11:0,
BUILD FAILED
D:\programming\lwjgl3\lwjgl3\build.xml:442: The following error occurred while executing this line:
D:\programming\lwjgl3\lwjgl3\config\android\build.xml:247: The following error occurred while executing this line:
D:\programming\lwjgl3\lwjgl3\config\android\build.xml:60: The following error occurred while executing this line:
D:\programming\lwjgl3\lwjgl3\config\android\build.xml:31: apply returned: 1
```

I checked some other issues like [this one](https://social.msdn.microsoft.com/Forums/vstudio/en-US/bb6a6876-426a-499c-ae52-64b8cd16b011/where-is-inttypesh?forum=vcgeneral), and it looks like the file has been removed completely. [This](https://github.com/json-c/json-c/issues/22) references the same thing, though [this SO post](https://stackoverflow.com/questions/13266868/ffmpeg-inttypes-h-not-found-error) with a download link. Downloading and adding the files to the project cause different exceptions, most notably that it has to be compiled in Visual Studio.

I can't use VS since the project is designed for IntelliJ. And there's (currently) no way to import INtelliJ projects into VS. Removing the throws cause different exceptions for different files that aren't found. 

**TL:DR;** `inttypes.h` doesn't exist anymore. How can I get it to compile?

**Comment by Spasi** [comment]
You're not supposed to use msvc to compile the Android branch. You must set the `ANDROID_SDK_HOME` environment variable to point to the Android SDK and LWJGL will use the NDK toolchain to build everything.

**Comment by LunarWatcher** [comment]
`ANDROID_SDK_HOME` is set, it points to the root where the NDK is installed in the default install location.  The NDK was used in the first place, my attempt on using a downloaded `inttypes.h` file was an attempt to get it to work. With or without the NDK, it doesn't compile.

**Comment by Spasi** [comment]
This could be an issue with NDK 16, which has the unified headers only. In my setup there are 64 different `inttypes.h` files, but I have not upgraded to 16 yet. I'll post again when I have time to test it.

**Comment by LunarWatcher** [comment]
Using the NDK archive I manually downloaded the previous NDK version (r15c). It's transferring into the NDK-bundle folder right now, I'll post again if I can get it to work with a different version. I'm going to try the previous three versions until I either get through all or find one that works

**Comment by LunarWatcher** [comment]
Using [r15c](https://developer.android.com/ndk/downloads/older_releases.html) works perfectly, it compiles now. Some of the other issues I had were related to sloppy environment variable setting *cough*

Revision 16 breaks the build, probably doe to the unified headers you mentioned.

**Comment by cainy393** [comment]
Just ran into this issue when compiling for android. First off retrolambda.jar was misssing from the lib directory but you can download that from [here](https://oss.sonatype.org/content/groups/public/net/orfjackal/retrolambda/retrolambda/) but once that was sorted I got the same error. Rolling back to r15c as @LunarWatcher suggested worked though.
